libvast/vast/format/make_writer.hpp

```diff
@@ -0,0 +1,62 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/config.hpp"
+#include "vast/detail/make_io_stream.hpp"
+#include "vast/format/writer.hpp"
+
+#if VAST_HAVE_PCAP
+#  include "vast/format/pcap.hpp"
+#endif
+
+#include <caf/expected.hpp>
+#include <caf/settings.hpp>
+
+#include <memory>
+#include <string>
+#include <type_traits>
+
+namespace vast::format {
+
+template <class Writer>
```

Comment:
This is missing documentation. A note on why this returns an `expected<unique_ptr<T>>` would be nice-to-have.